service/geopmdpy/session.py

```diff
@@ -31,91 +31,563 @@
+
+        Parse a user supplied stream into a list of tuples
+        representing read requests.  The tuples are of the form
+        (signal_name, domain_type, domain_idx) and are parsed one from
+        each line of the stream.  Each of the values in the stream is
+        separated by white space.
+
+        Each signal_name should match one of the signal names provided
+        by the service.  The domain_type is specified as the name
+        string, i.e one of the following strings: "board", "package",
+        "core", "cpu", "board_memory", "package_memory", "board_nic",
+        "package_nic", "board_accelerator", "package_accelerator".
+        The domain index is a positive integer indexing the specific
+        domain.
+
+        Args:
+            request_stream (file): Input stream to parse for read
+                                   requests
+
+        Returns:
+            list((str, int, int)): List of request tuples. Each
+                                   request comprises a signal name,
+                                   domain type, and domain index.
+
+        Raises:
+            RuntimeError: Line from stream does not split into three
+                          words and is also not a comment or empty
+                          line.
+
+        """
+        requests = []
+        for line in self.iterate_stream(request_stream):
+            words = line.split()
+            if len(words) != 3:
+                raise RuntimeError('Read request must be three words: "{}"'.format(line))
+            try:
+                signal_name = words[0]
+                domain_type = topo.domain_type(words[1])
+                domain_idx = int(words[2])
+            except (RuntimeError, ValueError):
+                raise RuntimeError('Unable to convert values into a read request: {}'.format(line))
+            requests.append((signal_name, domain_type, domain_idx))
+        return requests
+
+    def query_formats(self, signal_names, geopm_proxy):
+        """Call the GEOPM D-Bus API to get the format type for each signal name
+
+        Returns a list of geopm::string_format_e integers that determine
+        how each signal value is formatted as a string.
+
+        Args:
+            signal_names (list(str)): List of signal names to query.
+
+            geopm_proxy (dasbus.client.proxy.InterfaceProxy): The
+                dasbus proxy for the GEOPM D-Bus interface.
+
+        Returns:
+            list(int): List of geopm::string_format_e integers, one
+                       for each signal name in input list.
+
+        """
+
+        try:
+            result = [info[4] for info in
+                      geopm_proxy.PlatformGetSignalInfo(signal_names)]
+        except DBusError as ee:
+            if 'io.github.geopm was not provided' in str(ee):
+                err_msg = """The geopm systemd service is not enabled.
+    Install geopm service and run 'systemctl start geopm'"""
```

Comment:
```suggestion
                err_msg = """The geopm systemd service is not enabled.
                             Install geopm service and run 'systemctl start geopm'"""
```

Reply:
We don't want that much white space in the printed message, I thought a 4 space indent looks okay.

Reply:
What about this instead?
```suggestion
                err_msg = "The geopm systemd service is not enabled. " \
                          "Install geopm service and run 'systemctl start geopm'"
```